lib/drivers/mock/mock.go

```diff
@@ -6,21 +6,32 @@ import (
 
 type MockDriver struct {
 	Revisions []core.Revision
-	State     core.Revision
 }
 
 func (this *MockDriver) Init(config map[string]string) error {
 	return nil
 }
 
-func Migrate(r core.Revision, ec chan error) {
+func (this *MockDriver) Migrate(r core.Revision, ec chan error) {
+	for _, mig := range r.Migrate {
+		if _, err = mig; err != nil {
+			ec <- err
+			return
+		}
+	}
 }
 
-func Rollback(r core.Revision, ec chan error) {
+func (this *MockDriver) Rollback(r core.Revision, ec chan error) {
```

Comment:
You way over complicated this my man. Since this is a mock, we don't need to apply ANY rollback actions to the driver. We just need to pull the NEWEST (HEAD) revision out of the end of the array. and push NIL to EC